nvflare/fuel/utils/fobs/decomposers/via_file.py

```diff
@@ -68,45 +57,26 @@ def __init__(self):
         self.target_to_item = {}  # target_id => item_id
         self.target_items = {}  # item_id => item value
         self.last_item_id = 0
-        self.waiter = None
 
         # some stats
         self.file_creation_time = None
         self.file_size = 0
-        self.num_files = 0
-        self.num_secondary_msgs = 0
-        self.num_lookups = 0
-        self.lock = threading.Lock()
 
     def set_file_creation_time(self, duration):
         self.file_creation_time = duration
 
     def set_file_size(self, size):
         self.file_size = size
 
-    def inc_files(self):
-        with self.lock:
-            self.num_files += 1
-
-    def inc_2nd_msgs(self):
-        with self.lock:
-            self.num_secondary_msgs += 1
-
-    def inc_lookup(self):
-        with self.lock:
-            self.num_lookups += 1
-
     def add_item(self, item: Any):
         target_id = id(item)
         item_id = self.target_to_item.get(target_id)
-        if item_id:
-            return item_id
-
-        item_id = f"T{self.last_item_id}"
-        self.last_item_id += 1
-        self.target_items[item_id] = item
-        self.target_to_item[target_id] = item_id
-        return item_id
+        if not item_id:
+            item_id = f"T{self.last_item_id}"
+            self.last_item_id += 1
```

Comment:
This line is not thread-safe

Reply:
No thread issue here since FOBS processes all items sequentially for the same message.